React/Nextjs Project Post Setup Steps.md

```diff
@@ -0,0 +1,679 @@
+    ```
+    to either install `polyglot-cli` package globally or to update it. If installing for the first time, running
+     `polyglot login` is also required. 
+1. Run command
+    ```bash
+    polyglot init
+    ```
+   and type project number (if project isn't listed contact a team lead).
+   
+   Select options based on project requirements but for "*Relative path to the translations folder...*" enter:
+   ```bash
+   src/localisation/locales
+   ```
+   
+   After initialisation, there are 2 approaches for `localisation` and/or `locales` folder generation:
+   1. Go to `https://infinum.polyglothq.com/projects/[project_id]/translation_keys`, add a key and run 
+       ```bash
+       polyglot pull
+       ```
+       This will create `localisation/locales` with `json` files automatically.
+   1. Manually create `localisation` folder and `locales` folder  inside `localisation` and next time translations 
+   are pulled, locale files will be available.
+1. In `package.json` add following entry at the end of `scripts` property
+    ```bash
+    "polyglot:pull": "polyglot pull -c .polyglot.json"
+    ```
+1. In `localisation` folder create file `localisation.ts` and add following content
+    ```bash
+    import i18n from 'i18next';
+    import { initReactI18next } from 'react-i18next';
+    
+    import en from './locales/en_US.json';
+    
+    i18n.use(initReactI18next).init({
+        resources: {
+            // add new locales here; key names depend on Polyglot setup
+            'en_US': {
+                translations: en,
+            },
+        },
+        defaultNS: 'translations',
+        lng: 'en_US',
+        fallbackLng: 'en_US',
+        interpolation: {
+            escapeValue: false,
+        },
+    });
+
+    export default i18n;
+    ```
+1. In `_app.tsx` add following import
+    ```bash
+    import 'localisation/localisation';
+    ```
+
+Check if integration is correct by adding translated text in the UI.
+```tsx
+// example
+import React, { ReactElement } from 'react';
+import { useTranslation } from 'react-i18next';
+
+export const myComp = () => {
+  const { t } = useTranslation();
+  
+  return (
+    <div>
+      {t('some.key.registeredInPolyglot')}
+      ...
+    </div>
+  );
+};
+```
+
+Pulling translations is required when adding new keys in [Infinum Polyglot](https://infinum.polyglothq.com/).
+
+##### References:
+[i18n](https://github.com/mashpie/i18n-node)
+
+[i18n in Next.js](https://react.i18next.com/)
+
+[Polyglot CLI](https://github.com/infinum/js-polyglot-cli)
+
+[Infinum Polyglot](https://infinum.polyglothq.com/)
+
+
+### Theming
+##### Depends on sections: TypeScript
+
+| Package | Version |
+|---|---:|
+| @emotion/core | 10.0.x |
+| @emotion/styled | 10.0.x |
+| emotion-theming | 10.0.x |
+| emotion-normalize | 10.1.x |
+
+To add theming support do the following:
+1. Run command
+    ```bash
+    yarn add @emotion/core @emotion/styled emotion-theming emotion-normalize
```

Comment:
This should be updated for the chakra flow (or just link to the chakra chapter?)